datalab/mlalpha/_package_runner.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Implements running Datalab ML Solution Packages."""
+
+import inspect
+import google.cloud.ml as ml
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+
+PACKAGE_NAMESPACE = 'datalab_solutions.'
+
+class PackageRunner(object):
+  """A Helper class to run Datalab ML solution packages."""
+
+  def __init__(self, package_uri):
+    """
+    Args:
+      package_uri: The uri of the package. The file base name needs to be in the form of
+          "name-version", such as "inception-0.1". The first part split by "-" will be used
+          as the last part of the namespace. In the example above,
+          "datalab_solutions.inception" will be the namespace.
+    """
+    self._package_uri = package_uri
+    self._name = os.path.basename(package_uri).split('-')[0]
+    
+  def _install_to_temp(self):
```

Comment:
It's not horribly slow but it may be different if we add dependencies in setup.py.

It is important that each function call needs to be stateless. Otherwise it is really hard to manage the session and control when to clean up. Indeed the --usage can be further improved to call it once. Added todo for that.